test/e2e/e2e.go

```diff
@@ -158,6 +158,9 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 	// Reference federation test to make the import valid.
 	federationtest.FederationSuite = commontest.FederationE2E
 
+	// Run test suite init actions (if any) added. Called only once for test suite during setup.
+	framework.RunSuiteInitActions()
```

Comment:
Only who wish to register the hooks. some scenarios might call for it. It is a just a provision. I had one scenraio which might have utilized this facility for cluster registration to federation. But @madhusudancs think it can be achieved by other means.